static/js/stream_data.js

```diff
@@ -194,19 +194,17 @@ exports.render_stream_description = function (sub) {
 
 exports.update_calculated_fields = function (sub) {
     sub.is_admin = page_params.is_admin;
-    // Admin can change stream name/description either stream is public or
-    // stream is private and admin is subscribed to private stream.
-    sub.can_change_name_description = page_params.is_admin &&
-                                     (!sub.invite_only || (sub.invite_only && sub.subscribed));
+    // Admin can change any stream's name & description either stream is public or
+    // private, subscribed or unsubscribed.
+    sub.can_change_name_description = page_params.is_admin;
     // If stream is public then any user can subscribe. If stream is private then only
     // subscribed users can unsubscribe.
     sub.should_display_subscription_button = !sub.invite_only || sub.subscribed;
     sub.can_make_public = page_params.is_admin && sub.invite_only && sub.subscribed;
     sub.can_make_private = page_params.is_admin && !sub.invite_only;
     sub.can_change_subscription_type = sub.can_make_public || sub.can_make_private;
-    // User can access subscribers as well as add other users to stream
-    // if sub.can_add_subscribers is true.
-    sub.can_add_subscribers = !sub.invite_only || (sub.invite_only && sub.subscribed);
+    // User can add other users to stream if stream is public or user is subscribed to stream.
+    sub.can_access_subscribers = !sub.invite_only || sub.subscribed || page_params.is_admin;
```

Comment:
I suspect in a follow-up we're going to need to split out a `can_add_subscribers` variable to handle the specific UI widget for adding subscribers (Which realm admins still can't do, and we intend to keep that way).